src/ar/Tag.cpp

```diff
@@ -0,0 +1,153 @@
+#include "Tag.h"
+
+namespace AR
+{
+double findAngle(cv::Point a, cv::Point b, cv::Point c)
+{
+	a = a - b; // shift vectors so that B is at the origin
+	b = b - b;
+	c = c - b;
+	double dot_product = a.ddot(c);
+	double a_mag = sqrt((a.x * a.x) + (a.y * a.y));
+	double c_mag = sqrt((c.x * c.x) + (c.y * c.y));
+	return acos(dot_product / (a_mag * c_mag));
+}
+
+cv::Vec3d rotationMatrixToEulerAngles(cv::Mat &R)
+{
+	double sy = sqrt(R.at<double>(0, 0) * R.at<double>(0, 0) +
+	                 R.at<double>(1, 0) * R.at<double>(1, 0));
+
+	bool singular = sy < 1e-6;
```

Comment:
can we create an epsilon constant, or a function if applicable, for the fuzzy comparison